src/extension.ts

```diff
@@ -4,12 +4,17 @@ import * as fs from 'fs';
 import { addNote, addPlainNote } from './note-db';
 import { generateMarkdownReport } from './reporting';
 import { NotesTree, TreeActions } from './notes-tree';
-import { initializeStorageLocation, getAnnotationFilePath } from './configuration';
+import { initializeStorageLocation, getAnnotationFilePath, getConfiguration } from './configuration';
 
 export function activate(context: vscode.ExtensionContext) {
     console.log('Extension "code-annotation" is now active!');
 
-    initializeStorageLocation(context.globalStoragePath);
+    const configuration = getConfiguration();
+
+    // if path is '', use global.
+    const storagePath = !configuration.saveLocation ? context.globalStoragePath : configuration.saveLocation;
+
+    initializeStorageLocation(storagePath);
```

Comment:
I think this might lead to some errors in the future. A better approach would be to change to have a global variable in `configuration.ts` that holds the global storage path.  That would be initialized on `activate`.
With that, `getAnnotationFilePath` will actually query the configurations and check the configuration set by the user (in a way you already doing here).
This way, all manipulation is done only on `configuration.ts`, and nowhere else.
It's possible that the actual creation of the path might need to be done on `getAnnotationFilePath`.

What do you think?

Reply:
Great shout! 
I'll make the changes soon.